File: 2176. Count Equal and Divisible Pairs in an Array/countPairs.cpp
using namespace std;
#include <iostream>


class Solution {
public:
    int countPairs(vector<int>& nums, int k) {
        int sz = nums.size();
        int ans = 0;

        for(int i=0; i<sz-1; i++){

            for(int j= sz-1; j>i; j--){

                if(nums[i] == nums[j] && i*j % k  == 0){
                    ans++;
                }
            }
        }
        return ans;
    }
};

File: 2176. Count Equal and Divisible Pairs in an Array/countPairs.js
var countPairs = function (nums, k) {
    let len = nums.length
    let ans = 0

    for (let i = 0; i < len - 1; i++) {
        // if(nums.lastIndexOf(nums[i]) === i) continue;
        for (let j = len - 1; j > i; j--) {
            if (nums[i] === nums[j] && i * j % k === 0) {
                ans++
            }
        }
    }

    return ans
};